### PR TITLE
Handle ErrorKind::Interrupted when waiting for events

### PR DESCRIPTION
**Pull request by acatangiu**
## Description of Changes

ErrorKind::Interrupted (libc::EINTR) errors should not break the HttpServer requests loop, they are internally consumed and reported as no events rather than requests errors.

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license.

## PR Checklist

`[Author TODO: Meet these criteria.]`
`[Reviewer TODO: Verify that these criteria are met. Request changes if not]`

- [x] All commits in this PR are signed (`git commit -s`).
- [x] The reason for this PR is clearly provided (issue no. or explanation).
- [x] The description of changes is clear and encompassing.
- [x] Any required documentation changes (code and docs) are included in this PR.
- [x] Any newly added `unsafe` code is properly documented.
- [x] Any user-facing changes are mentioned in `CHANGELOG.md`.

### CODE REVIEW

**Review by andreeaflorescu — DISMISSED** [review]

**Comment by stappersg** [comment]
Coverage-x86 bails out with
> AssertionError: Coverage drops by 0.40%. Please add unit tests for the uncovered lines.

**Comment by acatangiu** [comment]
Updated coverage, tests are passing.

File: .gitignore [hunk 1]
```diff
@@ -0,0 +1,3 @@
+.idea
```
Comment (andreeaflorescu) [inline .gitignore]: Can you please remove `.idea` from gitignore? The best practice is to add the IDE specific files/dirs to your machine gitignore and not to the project. Each developer can use different IDEs so the gitignore would be clutter if everyone adds theirs.

You can also check this article: https://blog.martinhujer.cz/dont-put-idea-vscode-directories-to-projects-gitignore/

Comment (acatangiu) [reply .gitignore]: While I understand the concept, I think Rust only has a few IDEs. The trade-off is a few lines in gitignore vs having `git status` always report untracked files which is very annoying during development.

Adding stuff locally to `.gitignore` makes `git status` report that as changed. Then you'd have to add `.gitignore` itself to `.gitignore` to get rid of all the warnings.

That's why I thought the majority of other rust crates in this org and in rust-vmm have `.idea` in their `.gitignore`.

> https://blog.martinhujer.cz/dont-put-idea-vscode-directories-to-projects-gitignore/

I think this is should not be applied religiously, but on a case-by-case.

# TLDR
I will remove `.idea` if you feel strongly about it, but I think the pros outweigh the cons.

Comment (acatangiu) [reply .gitignore]: Scratch that, I just saw the global `.gitignore` option, that should work nicely.

Comment (andreeaflorescu) [reply .gitignore]: The global gitignore option is quite nice. The problem is more pressing for something like rust-vmm where there are quite a few crates, that's how I discovered the global gitignore as well :))

**Review by andreeaflorescu — APPROVED** [review]

File: src/server.rs [hunk 2]
```diff
@@ -301,8 +302,11 @@ impl HttpServer {
         // current thread until at least one event is received.
         // The received notifications will then populate the `events` array with
         // `event_count` elements, where 1 <= event_count <= MAX_CONNECTIONS.
-        let event_count =
-            epoll::wait(self.epoll_fd, -1, &mut events[..]).map_err(ServerError::IOError)?;
+        let event_count = match epoll::wait(self.epoll_fd, -1, &mut events[..]) {
+            Ok(event_count) => event_count,
+            Err(e) if e.kind() == std::io::ErrorKind::Interrupted => 0,
```
Comment (iulianbarbu) [inline src/server.rs:307]: nit: Add a comment which gives some insight why this handling is needed: "`epoll::wait` might be interrupted by signals and return with `libc::EINTR`. This does not break any `micro-http` functionality, and needs to return gracefully."

**Review by iulianbarbu — APPROVED** [review]

**Comment by jiangliu** [comment]
could someone help to merge this MR?

**Comment by acatangiu** [comment]
> could someone help to merge this MR?

@jiangliu done.
